quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -231,6 +230,18 @@ impl Storage for LocalFileStorage {
         })?
     }
 
+    #[tracing::instrument(skip(self), level = "debug")]
+    async fn get_slice_stream(
+        &self,
+        path: &Path,
+        range: Range<usize>,
+    ) -> StorageResult<Box<dyn AsyncRead + Send + Unpin>> {
+        let full_path = self.full_path(path)?;
+        let mut file = tokio::fs::File::open(&full_path).await?;
+        file.seek(SeekFrom::Start(range.start as u64)).await?;
```

Comment:
You ignored range.end. this is a bug.

`.take`

Reply:
Good catch, sorry for letting this one sleep through! When wanting to add a test for this I discovered the `storage_test_suite` that I missed until now!